### DISCUSSION TITLE
Send JSON String On API POST Body

### DISCUSSION THREAD

**Discussion by hci-ikahera** [Budibase/budibase#9905] [Help] [answered]
Hello everyone,
Anyone here can help me? I want to use the value that I got from JSON Field as a body of a REST API POST. Here is how I write it in REST API datasource.

<img width="985" alt="Screenshot 2023-03-07 at 13 45 34" src="https://user-images.githubusercontent.com/95213000/223345449-d73473af-8590-4650-a87a-a7538b86d803.png">

But everytime I execute the query, it always returns error as below,

<img width="768" alt="Screenshot 2023-03-07 at 13 49 08" src="https://user-images.githubusercontent.com/95213000/223345779-f87e8a3c-5deb-42c1-9158-d87378621a31.png">

I found out that when the value should be JSON string, for example **"{\"data1\":\"QWE\",\"data2\":\"QAZ\"}"**, it changes back to the form of JSON when the query is being executed, for example **{"data1": "QWE", "data2": "QAZ"}**

Anyone here has a solution for this problem?
Thank you.

Regards,
Ika

**Comment by andz-bb** [comment, marked as answer]
Hey @hci-ikahera 

I've had a play around with this on my end. Would you be able to try 
```json
{
  "payload": {{ payload }}
}
```
as the body for your query?

![SCR-20230307-ph2](https://user-images.githubusercontent.com/110921612/223514631-154e641f-ed5e-4695-8e20-b7393edd7a9f.png)



In my case I used `{ "hey": "hey" }` as my default and the query editor was happy with it.

<img width="1277" alt="image" src="https://user-images.githubusercontent.com/110921612/223514925-78eb8e31-0e22-47c9-9a93-ec35213aa236.png">


Here's how I implemented it in my app:
![SCR-20230307-pfd](https://user-images.githubusercontent.com/110921612/223515049-af10d923-1017-4d4e-87ae-4cdb329a1e50.png)
<img width="1683" alt="image" src="https://user-images.githubusercontent.com/110921612/223515187-80098d49-97fc-4edc-8705-5e123aee63d2.png">

When I run the action it successfully posts the content of my payload field using the payload binding. Here's what hit the API:
<img width="397" alt="image" src="https://user-images.githubusercontent.com/110921612/223515711-9f3d4281-56db-4a19-9ff9-6dec391e03cd.png">


If you're still having issues with this let me know and I can have a closer look. Hope this helps!

**Reply by hci-ikahera** [reply]
Thanks for your help. It works for me 🙏🏼
